### DISCUSSION TITLE
Sonoff DualR3 & Homekit

### DISCUSSION THREAD

**Discussion by andreabaroni** [indomus/forum#677] [GENERALE - Discussioni generiche]
Buonasera, sono giorni che sto cercando diverse guide in merito alla domotica legata alla tapparelle. In particolare mi piacerebbe utilizzare il Sonoff DualR3, che mi sembra una valida opzione.

Quello che non riesco a capire è se è possibile un integrazione con Homekit. Devo utilizzare per forza Home Assistant per fare da "Bridge" ?

**Comment by snakuzzo** [comment]
Puoi usare il firmware [Ravencore](https://github.com/RavenSystem/esp-homekit-devices) in alternativa.
Ne abbiamo parlato [QUI](https://indomus.it/componenti/home-accessory-architect-haa-o-ravencore-v2/)

ma dovresti dare un'occhiata alla documentazione del progetto prima

**Reply by andreabaroni** [reply]
Grazie mille, visto che è stato molto celere..chiedo anche un'altra cosa. Se invece faccio integrazione con Home Assistant, ha idea di come venga visualizzato il DualR3 ? come 2 switch ? o riconosce che si tratta di una tapparella ?

Ancora grazie mille

**Reply by snakuzzo** [reply]
Dipende da come esponi (diamoci del tu :) ) il device e dalla `device_class` che dai all'entità.

Se crei un'entità di tipo `cover` verrà vista come tapparella (in questo caso guarda anche [QUI](https://www.home-assistant.io/integrations/alexa.smart_home/#cover))
Se crei due entità di tipo `switch` verranno viste come due interruttori

**Reply by andreabaroni** [reply]
Grazie mille! Tutto chiaro!
